### ISSUE TITLE
[Metricbeat] ActiveMQ module

### ISSUE THREAD

**Issue by mtojek** [elastic/beats#14510]
Here is the tracking issue for the ActiveMQ metricbeat module development.

# Introduction

<details><summary>show/hide</summary>

ActiveMQ is a Java-based messaging server. Client applications can be integrated using the following protocols: AQMP, STOMP, MQTT and JMS.

> Starting with ActiveMQ 5.8.0 Jolokia is bundled which provides a RESTful interface to ActiveMQ’s JMX capabilities.

(source: https://activemq.apache.org/how-can-i-monitor-activemq)

The module should collect metrics for the broker, all queues and topics.

</details>

# Metricbeat Module / Dataset release checklist

This checklist is intended for Devs which create or update a module to make sure modules are consistent.

## Modules

For a metricset to go GA, the following criterias should be met:

* [x] Supported versions are documented
* [ ] ~~Supported operating systems are documented (if applicable)~~
* [ ] ~~Integration tests exist~~
* [x] System tests exist
* [x] Automated checks that all fields are documented
* [x] Documentation
* [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
* [x] Dashboards exists (if applicable)
* [x] Kibana Home Tutorial (if applicable)
   * [x] Open issue in [EUI repo](https://github.com/elastic/eui) to add [icon for module](https://elastic.github.io/eui/#/display/icons) if not already exists.
* [x] Open PR against Kibana repo with tutorial. Examples can be found [here](https://github.com/elastic/kibana/tree/master/src/legacy/core_plugins/kibana/server/tutorials).
* [ ] Script in configuration is moved to `manifest.yml` https://github.com/elastic/beats/issues/14742

## Metricbeat module

* [ ] ~~Example `data.json` exists and an automated way to generate it exists (`go test -data`)~~
* [x] Test environment in Docker exist for ~~integration~~ system tests

# Metrics

This is the list of sample metrics that the server should expose - consider them as basic requirements. 

<details><summary>show/hide</summary>

## Broker

- MemoryPercentUsage - memory percent usage
- StorePercentUsage - store percent usage
- TempPercentUsage - temporary percent usage
- TotalConnectionsCount - total connections count
- TotalConsumerCount - total consumer count
- TotalDequeueCount - total dequeue count
- TotalEnqueueCount - total enqueue count
- TotalMessageCount - total message count
- TotalProducerCount - total producer count

## Queue

- AverageEnqueueTime - average enqueue time
- AverageMessageSize - average message size
- ConsumerCount - consumer count
- DequeueCount - dequeue count
- DispatchCount - dispatch count
- EnqueueCount - enqueue count
- ExpiredCount - xxpired count
- InFlightCount - in-flight count
- MaxEnqueueTime - max enqueue Time
- MemoryPercentUsage - memory percent usage
- MinEnqueueTime - min enqueue Time
- ProducerCount - producer count
- QueueSize - queue size

## Topic

- AverageEnqueueTime - average enqueue time
- AverageMessageSize - average message size
- ConsumerCount - consumer count
- DequeueCount - dequeue count
- DispatchCount - dispatch count
- EnqueueCount - enqueue count
- ExpiredCount - xxpired count
- InFlightCount - in-flight count
- MaxEnqueueTime - max enqueue Time
- MemoryPercentUsage - memory percent usage
- MinEnqueueTime - min enqueue Time
- ProducerCount - producer count

</details>

**Comment by mtojek** [comment]
FYI @exekias

**Comment by exekias** [comment]
Great to know that Jolokia is now bundled! that makes using our jolokia metricset even easier. Have a look to `tomcat` module for an example on how to leverage it here :)

**Comment by mtojek** [comment]
Agree ;) I believe this will be the first supported version then.

btw. Thanks for adding tags!

**Comment by mtojek** [comment]
The PR with module implementation has been merged. Also, the checklist above has been updated.

**Comment by mtojek** [comment]
I noticed an issue with missing Python dependency in Windows environments. After a short investigation, I've raised a GH issue to the Infra team.

**Comment by exekias** [comment]
ey @mtojek I just saw your issue to get the deps. We don't really run integration tests on windows, due to the requirement limitations, among other things. Your test would actually be skiped by

```
@unittest.skipUnless(metricbeat.INTEGRATION_TESTS, "integration test")
```

The problem is that stomp is imported before that. What we normally do is importing extra libraries inside called methods, and not globally. See https://github.com/elastic/beats/blob/master/metricbeat/tests/system/test_kafka.py#L50 as an example.

I think you can do this change and close the infra issue if you want

**Comment by mtojek** [comment]
Didn't know about that, will fix this in sec. Thanks!

**Comment by mtojek** [comment]
Due to a different scale of Memory Usage related fields (pct: 1..100 vs 0..1), there is a dedicated PR https://github.com/elastic/beats/pull/14711#pullrequestreview-321585855 to convert the value to the appropriate range.

**Comment by mtojek** [comment]
The PR with dashboards has been merged. I'm proceeding with updating with Kibana updates (home, icon, etc.)

**Comment by mtojek** [comment]
The PR with Kibana tutorial has been merged. Backport will be delivered as soon as the branch _7.x_ in elastic/beats is merged.

**Comment by mtojek** [comment]
Metricbeat module development has been finished. Remaining work (processor definitions) will be adressed in https://github.com/elastic/beats/issues/14742 .

Resolving

**Comment by ChrsMark** [comment]
Regression found while testing with BC1:
```
Error creating runner from config: 1 error: the processor action script does not exist. Valid actions: decode_json_fields, rename, community_id, include_fields, truncate_fields, add_locale, add_observer_metadata, dns, copy_fields, decode_base64_field, drop_fields, extract_array, add_docker_metadata, add_kubernetes_metadata, dissect, add_tags, decompress_gzip_field, add_cloud_metadata, add_host_metadata, add_process_metadata, convert, registered_domain, add_fields, add_labels, drop_event
```


The script processor introduced at https://github.com/elastic/beats/pull/14655 was not included in `7.6` branch after https://github.com/elastic/beats/pull/14711 was merged.
We need to backport https://github.com/elastic/beats/pull/14711 to `7.6` so as the module to be functional.

cc: @mtojek @exekias

**Comment by ChrsMark** [comment]
Tested with BC2 and works as expected 👌 !
